app/src/main/java/com/portfolio/tasky/TaskyValidationWatcherImpl.kt

```diff
@@ -0,0 +1,64 @@
+package com.portfolio.tasky
+
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.appcompat.widget.AppCompatEditText
+import com.portfolio.tasky.globals.Constants.Companion.Entry.LOGIN
+import com.portfolio.tasky.globals.Constants.Companion.Entry.REGISTRATION
+import com.portfolio.tasky.globals.DataUtils.Companion.Validators.isEmailValid
+import com.portfolio.tasky.globals.DataUtils.Companion.Validators.isNameValid
+import com.portfolio.tasky.globals.DataUtils.Companion.Validators.isPasswordValid
+import com.portfolio.tasky.views.TaskyAppCompatEditText
+
+class TaskyValidationWatcherImpl(
+    taskyAppCompatEditText: TaskyAppCompatEditText,
+    entry : String,
+    fieldsValidator : FieldsValidator
+) : TextWatcher {
+
+    private var appCompatEditText: AppCompatEditText
+    var taskAppCompatEditText: TaskyAppCompatEditText = taskyAppCompatEditText
+    private var isEntryValid : String = ""
+    var fieldsValidator : FieldsValidator? = null
+
+    init {
+        this.appCompatEditText = taskyAppCompatEditText.subLayout.etInput
+        this.isEntryValid = entry
+        this.fieldsValidator = fieldsValidator
+    }
+
+    override fun beforeTextChanged(p0: CharSequence, p1: Int, p2: Int, p3: Int) {/*TODO("Not yet implemented")*/}
+
+    override fun onTextChanged(p0: CharSequence, p1: Int, p2: Int, p3: Int) {
+        validateText(p0, taskAppCompatEditText.id, taskAppCompatEditText)
+    }
+
+    override fun afterTextChanged(p0: Editable?) {/*TODO("Not yet implemented")*/}
+
+    private fun validateText(var1: CharSequence, viewID: Int, taskAppCompatEditText: TaskyAppCompatEditText) {
+        if(viewID == R.id.et_email){
```

Comment:
You are hardcoding the ids on the validator

What would happen if for example

You add this validator on Registration? Where they require a name
Or when you add a new attendee?

This means you'll have to always remember to add the id in this method

What I recommend, is having specific usecases
Ex: ValidateEmailUseCase, ValidatePasswordUseCase
And call it from a viewmodel
This means, everytime you have to use an email, you can just pass the EmailValidator, and that's it
No need to use IDs or anything

Reply:
True..

Done